### PR TITLE
Add initial version of BAMOE bom

### PR DESCRIPTION
**Pull request by baldimir**
- First draft version of BAMOE bom. It is based on kogito-bom. 
- Please review two things: 
    - I removed artifacts I thought can be removed. Please check if I didn't remove anything we want in the bom - you can do that by checking individual commits. I tried to name them appropriately. 
    - I expect there are other artifacts that can be removed, e.g. I was not 100% sure what can be removed from the process artifacts. Please comment those that I should remove.

### CODE REVIEW

File: bamoe-bom/pom.xml [hunk 1]
```diff
@@ -0,0 +1,1216 @@
...
+        <groupId>org.kie.kogito</groupId>
+        <artifactId>kogito-addons-quarkus-camel-deployment</artifactId>
+        <version>${version.org.kogito}</version>
+      </dependency>
+      <dependency>
+        <groupId>org.kie.kogito</groupId>
+        <artifactId>kogito-addons-quarkus-camel-deployment</artifactId>
+        <version>${version.org.kogito}</version>
+        <classifier>sources</classifier>
+      </dependency>
```
Comment (porcelli) [inline bamoe-bom/pom.xml:150]: I don't think we need this for 9.0.0

File: bamoe-bom/pom.xml [hunk 2]
```diff
@@ -0,0 +1,1216 @@
...
+        <groupId>org.kie.kogito</groupId>
+        <artifactId>kogito-addons-quarkus-kubernetes-deployment</artifactId>
+        <version>${version.org.kogito}</version>
+      </dependency>
+      <dependency>
+        <groupId>org.kie.kogito</groupId>
+        <artifactId>kogito-addons-quarkus-kubernetes-deployment</artifactId>
+        <version>${version.org.kogito}</version>
+        <classifier>sources</classifier>
+      </dependency>
```
Comment (porcelli) [inline bamoe-bom/pom.xml:249]: same case for the kubernetes-services dependency

File: bamoe-bom/pom.xml [hunk 3]
```diff
@@ -0,0 +1,1216 @@
...
+        <groupId>org.kie.kogito</groupId>
+        <artifactId>kogito-addons-quarkus-kubernetes-test-utils</artifactId>
+        <version>${version.org.kogito}</version>
+      </dependency>
+      <dependency>
+        <groupId>org.kie.kogito</groupId>
+        <artifactId>kogito-addons-quarkus-kubernetes-test-utils</artifactId>
+        <version>${version.org.kogito}</version>
+        <classifier>sources</classifier>
+      </dependency>
```
Comment (porcelli) [inline bamoe-bom/pom.xml:260]: this is set of testing utils; maybe force the test scope? dunno

File: bamoe-bom/pom.xml [hunk 4]
```diff
@@ -0,0 +1,1216 @@
...
+        <groupId>org.kie.kogito</groupId>
+        <artifactId>kogito-timer</artifactId>
+        <version>${version.org.kogito}</version>
+      </dependency>
+      <dependency>
+        <groupId>org.kie.kogito</groupId>
+        <artifactId>kogito-timer</artifactId>
+        <version>${version.org.kogito}</version>
+        <classifier>sources</classifier>
+      </dependency>
```
Comment (porcelli) [inline bamoe-bom/pom.xml:307]: this is related to job scheduler, not sure if we need it.

File: bamoe-bom/pom.xml [hunk 5]
```diff
@@ -0,0 +1,1216 @@
...
+        <groupId>org.kie.kogito</groupId>
+        <artifactId>kogito-addons-quarkus-monitoring-elastic-deployment</artifactId>
+        <version>${version.org.kogito}</version>
+      </dependency>
+      <dependency>
+        <groupId>org.kie.kogito</groupId>
+        <artifactId>kogito-addons-quarkus-monitoring-elastic-deployment</artifactId>
+        <version>${version.org.kogito}</version>
+        <classifier>sources</classifier>
+      </dependency>
```
Comment (porcelli) [inline bamoe-bom/pom.xml:489]: not sure about elastic - thoughts?

File: bamoe-bom/pom.xml [hunk 6]
```diff
@@ -0,0 +1,1216 @@
...
+        <groupId>org.kie.kogito</groupId>
+        <artifactId>kogito-addons-quarkus-events-mongodb-deployment</artifactId>
+        <version>${version.org.kogito}</version>
+      </dependency>
+      <dependency>
+        <groupId>org.kie.kogito</groupId>
+        <artifactId>kogito-addons-quarkus-events-mongodb-deployment</artifactId>
+        <version>${version.org.kogito}</version>
+        <classifier>sources</classifier>
+      </dependency>
```
Comment (porcelli) [inline bamoe-bom/pom.xml:558]: I hope we don't need this

File: bamoe-bom/pom.xml [hunk 7]
```diff
@@ -0,0 +1,1216 @@
...
+        <groupId>org.kie.kogito</groupId>
+        <artifactId>kogito-addons-quarkus-task-notification-deployment</artifactId>
+        <version>${version.org.kogito}</version>
+      </dependency>
+      <dependency>
+        <groupId>org.kie.kogito</groupId>
+        <artifactId>kogito-addons-quarkus-task-notification-deployment</artifactId>
+        <version>${version.org.kogito}</version>
+        <classifier>sources</classifier>
+      </dependency>
```
Comment (porcelli) [inline bamoe-bom/pom.xml:681]: we don't have human tasks supported for 9.0.0 - I think we should try to remove it.

@pefernan we may have a problem with DevUI that ideally should not display anything related to tasks if no tasks are added. This is probably not how it's working today.

File: bamoe-bom/pom.xml [hunk 8]
```diff
@@ -0,0 +1,1216 @@
...
+        <groupId>org.kie.kogito</groupId>
+        <artifactId>kogito-jq-expression</artifactId>
+        <version>${version.org.kogito}</version>
+      </dependency>
+      <dependency>
+        <groupId>org.kie.kogito</groupId>
+        <artifactId>kogito-jq-expression</artifactId>
+        <version>${version.org.kogito}</version>
+        <classifier>sources</classifier>
+      </dependency>
```
Comment (porcelli) [inline bamoe-bom/pom.xml:986]: I don't think we need this - those are SWF specifics

File: bamoe-bom/pom.xml [hunk 9]
```diff
@@ -0,0 +1,1216 @@
...
+        <groupId>org.kie.kogito</groupId>
+        <artifactId>runtime-tools-quarkus-extension-deployment</artifactId>
+        <version>${version.org.kogito}</version>
+      </dependency>
+      <dependency>
+        <groupId>org.kie.kogito</groupId>
+        <artifactId>runtime-tools-quarkus-extension-deployment</artifactId>
+        <version>${version.org.kogito}</version>
+        <classifier>sources</classifier>
+      </dependency>
```
Comment (porcelli) [inline bamoe-bom/pom.xml:1143]: back to my previous comment, would be possible to have task split from the Dev UI?

File: bamoe-bom/pom.xml (hunk 7 above)
Comment (pefernan) [reply bamoe-bom/pom.xml:681]: Good question @porcelli. Currently *DevUi* always show the Task related pages, but it has a setting to configure which pages should be shown in the ui. We can just remove them from the list of available pages and add them back later or we can make a *smarter* mechanism to show and hide them.

File: bamoe-bom/pom.xml (hunk 3 above)
Comment (baldimir) [reply bamoe-bom/pom.xml:260]: I think these are part of the artifacts from your previous comment. So I think those are also not needed.

File: bamoe-bom/pom.xml (hunk 4 above)
Comment (baldimir) [reply bamoe-bom/pom.xml:307]: @elguardian Enrique, do you think we need it in BAMOE bom for 9.0, please?

File: bamoe-bom/pom.xml (hunk 5 above)
Comment (baldimir) [reply bamoe-bom/pom.xml:489]: I think some kind of monitoring should be good to have. I already saw a question about such thing on the community chat. What do you think? The description of the feature is here https://github.com/kiegroup/kogito-runtimes/tree/main/addons/common/monitoring

File: bamoe-bom/pom.xml (hunk 6 above)
Comment (baldimir) [reply bamoe-bom/pom.xml:558]: Based on https://github.com/kiegroup/kogito-runtimes/tree/main/addons/common/events it looks like a separate implementation. I agree that hopefully we don't need this :) I will remove it.

File: bamoe-bom/pom.xml [hunk 10]
```diff
@@ -0,0 +1,1216 @@
...
+        <groupId>org.kie.kogito</groupId>
+        <artifactId>jobs-service-api</artifactId>
+        <version>${version.org.kogito}</version>
+      </dependency>
+      <dependency>
+        <groupId>org.kie.kogito</groupId>
+        <artifactId>jobs-service-api</artifactId>
+        <version>${version.org.kogito}</version>
+        <classifier>sources</classifier>
+      </dependency>
```
[the remark this answers is not in the document]
Comment (baldimir) [reply bamoe-bom/pom.xml]: Removed.

File: bamoe-bom/pom.xml (hunk 1 above)
Comment (baldimir) [reply bamoe-bom/pom.xml]: Removed.

File: bamoe-bom/pom.xml (hunk 2 above)
Comment (baldimir) [reply bamoe-bom/pom.xml]: Removed.

File: bamoe-bom/pom.xml (hunk 3 above)
Comment (baldimir) [reply bamoe-bom/pom.xml]: Removed.

File: bamoe-bom/pom.xml (hunk 6 above)
Comment (baldimir) [reply bamoe-bom/pom.xml:478]: Removed.

File: bamoe-bom/pom.xml (hunk 7 above)
Comment (baldimir) [reply bamoe-bom/pom.xml]: Removed.

File: bamoe-bom/pom.xml (hunk 8 above)
Comment (baldimir) [reply bamoe-bom/pom.xml]: Removed.

File: bamoe-bom/pom.xml [hunk 11]
```diff
@@ -0,0 +1,1059 @@
...
+        <groupId>org.kie.kogito</groupId>
+        <artifactId>kogito-addons-quarkus-mail-deployment</artifactId>
+        <version>${version.org.kogito}</version>
+      </dependency>
+      <dependency>
+        <groupId>org.kie.kogito</groupId>
+        <artifactId>kogito-addons-quarkus-mail-deployment</artifactId>
+        <version>${version.org.kogito}</version>
+        <classifier>sources</classifier>
+      </dependency>
```
Comment (martinweiler) [inline bamoe-bom/pom.xml:579]: I don't think we need the mail add on for STP

File: bamoe-bom/pom.xml [hunk 12]
```diff
@@ -0,0 +1,1059 @@
...
+        <groupId>org.kie.kogito</groupId>
+        <artifactId>kogito-addons-quarkus-rest-exception-handler</artifactId>
+        <version>${version.org.kogito}</version>
+      </dependency>
+      <dependency>
+        <groupId>org.kie.kogito</groupId>
+        <artifactId>kogito-addons-quarkus-rest-exception-handler</artifactId>
+        <version>${version.org.kogito}</version>
+        <classifier>sources</classifier>
+      </dependency>
```
Comment (martinweiler) [inline bamoe-bom/pom.xml:601]: Accordingto https://github.com/kiegroup/kogito-runtimes/tree/main/addons/common/rest-exception-handler this add on is used by others internally. Do we need to explicitly list it here?

Comment (baldimir) [reply bamoe-bom/pom.xml:601]: We need to do it as it needs to be pulled as a dependency in our specific version.

File: bamoe-bom/pom.xml [hunk 13]
```diff
@@ -0,0 +1,1059 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+  <parent>
+    <groupId>com.ibm</groupId>
```
Comment (porcelli) [inline bamoe-bom/pom.xml:7]: com.ibm.bamoe for groupId

**Review by porcelli — CHANGES_REQUESTED** [review]

File: bamoe-bom/pom.xml (hunk 5 above)
Comment (porcelli) [reply bamoe-bom/pom.xml:409]: Monitoring for sure, just would prefer to have prometheus only, at least for 9.0.0

Comment (baldimir) [reply bamoe-bom/pom.xml]: Removed.

File: bamoe-bom/pom.xml (hunk 4 above)
Comment (baldimir) [reply bamoe-bom/pom.xml:227]: Based on conversation with Enrique, we need this as it is mandatory in jbpm-flow project.

File: bamoe-bom/pom.xml (hunk 13 above)
Comment (baldimir) [reply bamoe-bom/pom.xml:7]: Fixed.

File: bamoe-bom/pom.xml (hunk 9 above)
Comment (baldimir) [reply bamoe-bom/pom.xml:920]: @porcelli @pefernan Is the response here enough please? Is it possible to remove these? https://github.com/IBM/bamoe-maven/pull/1#discussion_r1178837570

File: bamoe-bom/pom.xml (hunk 11 above)
Comment (baldimir) [reply bamoe-bom/pom.xml:513]: I checked with Enrique and this should be also used in reporting exceptions from REST, so should stay in. Is that ok please?

Comment (baldimir) [reply bamoe-bom/pom.xml]: Removed.

File: bamoe-bom/pom.xml (hunk 9 above)
Comment (porcelli) [reply bamoe-bom/pom.xml:898]: I don't think we'll be able to remove it; I hope we can tweak somehow to hide the task side.

File: pom.xml [hunk 14]
```diff
@@ -0,0 +1,24 @@
...
+
+    <groupId>com.ibm.bamoe</groupId>
+    <artifactId>bamoe-maven</artifactId>
+    <version>9.0-SNAPSHOT</version>
+    <packaging>pom</packaging>
+
+    <name>BAMOE product Maven modules parent</name>
+
+    <properties>
+        <maven.compiler.source>17</maven.compiler.source>
```
Comment (porcelli) [inline pom.xml:15]: I'm not sure if we should push to 17.

I'd tackle the 17 from the docs perspective, but I'd be on 11 here

Comment (baldimir) [reply pom.xml]: Fixed.

**Pull request merged**